client/template.go

```diff
@@ -131,18 +131,31 @@ func (payload *TemplateCreatePayload) Invalidate() error {
 		return errors.New("can't define terragrunt version for non-terragrunt template")
 	}
 
-	if payload.Type == TERRAGRUNT && payload.TerragruntVersion == "" {
-		return errors.New("must supply terragrunt version")
+	if payload.Type == TERRAGRUNT {
+		if payload.TerragruntVersion == "" {
+			return errors.New("must supply terragrunt version")
+		}
+
+		// The provider implicitly defaults to "opentofu".
+		if payload.TerragruntTfBinary == "" {
+			payload.TerragruntTfBinary = OPENTOFU
+		}
+
+		if payload.TerragruntTfBinary == OPENTOFU && payload.OpentofuVersion == "" {
```

Comment:
enforce opentofu or terraform versions (when applicible) in the provider side.
(This is a nice to have).